### DISCUSSION TITLE
2023-09-18 정기회의

### DISCUSSION THREAD

**Discussion by scarf005** [jiphyeonjeon-42/backend#775] [General]
이전 회의: #769

일정상 이유로 이번 주 회의는 온-오프라인 회의를 생략하겠습니다. 그 대신 각자 과제를 18일 전으로 완성하도록 합시다!

**Comment by scarf005** [comment]
# @Jiwon-Woo 과제

- https://zod.dev
- https://ts-rest.com
- https://tanstack.com/query/latest

공부하기

**Comment by scarf005** [comment]
@scarf005 과제

복잡한 쿼리 반환값 ID로 가져올때 invalidation 어떻게 할지 조사

**Reply by scarf005** [reply]
조사 결과, id만 돌려주는 방식은 프론트에 지나치게 복잡도를 전가시키는 문제가 있었습니다. 때문에, 기존 방식대로 진행하되, 추가 검색을 위해 id를 돌려주는 방식을 전체적으로 통일하면 될 것 같습니다.

예시) `/books/{id}`의 경우
### 기존
```json
{
	bookId: ...
	bookMetasId: ...
	book: ...
	<book 필드>...
	<bookMetas 필드>...
	lendings: [
		{ userId: ..., nickName: ..., history: [ ... ]
	]
}
```

### 개선안
```json
{
	book: {
		id: ...
		<book 필드>...
	},
	bookMetas: {
		id: ...
		<bookMetas 필드>...
	}
}
```

**Comment by scarf005** [comment]
@nyj001012 @honeyl3ee 
1. https://github.com/jiphyeonjeon-42/backend/blob/6cbb823e990591f1ab6cd6c9947388c4afb9d66d/backend/src/kysely/generated.ts#L35 를 기반으로 zod 스키마 만들기 
	(프론트엔드에서 쓰기 어려운 필드는 바꿔야됨 예시:
https://github.com/jiphyeonjeon-42/backend/blob/6cbb823e990591f1ab6cd6c9947388c4afb9d66d/backend/src/kysely/generated.ts#L61
같은 경우는 리터럴 유니언으로 변경
2. 기존 반환값들을 zod 기능을 활용해 중복을 최대한 줄이기
예: PATCH body로 `bookSchema.omit({ id: true }).partial()` 사용

**Comment by jimin52** [comment]
멤바가 되었습니다. 늦었지만 다시 참여하도록 하겠습니다.
근무시간은 평일 오전입니다
